src/modules/sensors/sensors.cpp

```diff
@@ -159,10 +161,14 @@ bool Sensors::init()
 
 int Sensors::parameters_update()
 {
-	if (_armed) {
+	if (_armed && !_failure_detector_updated) {
 		return 0;
 	}
 
+	if (_failure_detector_updated) {
```

Comment:
Fake sensors should be updated only if in HITL mode, I suppose? So maybe add a check here

Reply:
On the one hand, we change `_failure_detector_updated` only in those sections where we have already checked for HITL mode, on the other hand, adding a check can help in future code updates.